src/tls/Ticket.fst

```diff
@@ -155,15 +156,48 @@ let serialize t =
     | Ticket13 cs _ _ rms -> TLS_1p3, cs, rms in
   (versionBytes pv) @| (cipherSuiteBytes cs) @| (vlbytes 2 b)
 
-let create_ticket t =
-  let Key tid wr _ = !ticket_enc in
-  let plain = serialize t in
+let ticket_encrypt plain =
+  let Key tid wr _ = get_ticket_key () in
   let nb = CC.random (AE.iv_length tid) in
   let salt = AE.salt_of_state wr in
   let iv = AE.coerce_iv tid (xor 12ul nb salt) in
   let ae = AE.encrypt #tid #(length plain) wr iv empty_bytes plain in
   nb @| ae
 
+let create_ticket t =
+  let plain = serialize t in
+  ticket_encrypt plain
+
+let create_cookie (hrr:HandshakeMessages.hrr) (digest:bytes) (extra:bytes) =
```

Comment:
The RFC allows for cookies up to 2^16-1 bytes long. Depending on the length `of `extra` and `hrr` this function may create a longer cookie whose length won't fit in 2 bytes. Proving that we do not exceed this bound statically is going to be tricky.

Reply:
A solution is to add a variant of helloRetryRequestBytes that require hrr not to have a cookie extension and prove that the bound for this is in fact much tighter than 2^16 (assuming that the only extension is the protocol versions we can probably get a 2^8 bound). Then we can probably give 2^14 bytes or 2^15 bytes for the application data.